extension/memorylimiterextension/README.md

```diff
@@ -0,0 +1,105 @@
+will no longer be refused and no forced garbage collection will be performed).
+
+The difference between the soft limit and hard limits is defined via `spike_limit_mib`
+configuration option. The value of this option should be selected in a way that ensures
+that between the memory check intervals the memory usage cannot increase by more than this
+value (otherwise memory usage may exceed the hard limit - even if temporarily).
+A good starting point for `spike_limit_mib` is 20% of the hard limit. Bigger
+`spike_limit_mib` values may be necessary for spiky traffic or for longer check intervals.
+
+Note that while the extension can help mitigate out of memory situations,
+it is not a replacement for properly sizing and configuring the
+collector. Keep in mind that if the soft limit is crossed, the collector will
+return errors to all receive operations until enough memory is freed. This may
+eventually result in dropped data since the receivers may not be able to hold back
+and retry the data indefinitely.
+
+Please refer to [memoryli.go](../../internal/memorylimiter/memorylimiter.go) for the config spec.
+
+The following configuration options **must be changed**:
+- `check_interval` (default = 0s): Time between measurements of memory
+usage. The recommended value is 1 second.
+If the expected traffic to the Collector is very spiky then decrease the `check_interval`
+or increase `spike_limit_mib` to avoid memory usage going over the hard limit.
+- `limit_mib` (default = 0): Maximum amount of memory, in MiB, targeted to be
+allocated by the process heap. Note that typically the total memory usage of
+process will be about 50MiB higher than this value.  This defines the hard limit.
+- `spike_limit_mib` (default = 20% of `limit_mib`): Maximum spike expected between the
+measurements of memory usage. The value must be less than `limit_mib`. The soft limit
+value will be equal to (limit_mib - spike_limit_mib).
+The recommended value for `spike_limit_mib` is about 20% `limit_mib`.
+- `limit_percentage` (default = 0): Maximum amount of total memory targeted to be
+allocated by the process heap. This configuration is supported on Linux systems with cgroups
+and it's intended to be used in dynamic platforms like docker.
+This option is used to calculate `memory_limit` from the total available memory.
+For instance setting of 75% with the total memory of 1GiB will result in the limit of 750 MiB.
+The fixed memory setting (`limit_mib`) takes precedence
+over the percentage configuration.
+- `spike_limit_percentage` (default = 0): Maximum spike expected between the
+measurements of memory usage. The value must be less than `limit_percentage`.
+This option is used to calculate `spike_limit_mib` from the total available memory.
+For instance setting of 25% with the total memory of 1GiB will result in the spike limit of 250MiB.
+This option is intended to be used only with `limit_percentage`.
```

Comment:
Instead of copy-pasting the docs from the processor docs, I think we should keep it short with references to the processor for now. Just mention that this is the extension that will potentially replace the memory_limiter. It provides better guarantees from running out of memory as it will be used by the receivers to reject requests (or skip scraping) before converting them into OTLP. Mention that the configuration interface is the same as in the processor and that exporter is under development and does nothing yet.